exporter/datadogexporter/config.go

```diff
@@ -390,24 +391,11 @@ type HostMetadataConfig struct {
 	Tags []string `mapstructure:"tags"`
 }
 
-// LimitedTLSClientSetting is a subset of TLSClientSetting, see LimitedClientConfig for more details
-type LimitedTLSClientSettings struct {
-	// InsecureSkipVerify controls whether a client verifies the server's
-	// certificate chain and host name.
-	InsecureSkipVerify bool `mapstructure:"insecure_skip_verify"`
-}
-
-type LimitedClientConfig struct {
-	TLSSetting LimitedTLSClientSettings `mapstructure:"tls,omitempty"`
-}
-
 // Config defines configuration for the Datadog exporter.
 type Config struct {
-	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
-	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
-	configretry.BackOffConfig      `mapstructure:"retry_on_failure"`
-
-	LimitedClientConfig `mapstructure:",squash"`
+	confighttp.ClientConfig      `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
```

Comment:
I am wondering if we should take this opportunity to put this into a separate section instead of squashing everything; we are introducing 17 new options into the global namespace. We need to preserve compatibility with the old options (`tls`), but we can do that with a custom unmarshaling function

Reply:
For `confighttp.ClientConfig`, it seems the common practice in collector is to squash it in the top level in receiver/exporter configs, examples
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e2fb4ef305fb2f0de1e36f97e40b1ac0c33fdb4f/receiver/nsxtreceiver/config.go#L22
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e2fb4ef305fb2f0de1e36f97e40b1ac0c33fdb4f/exporter/lokiexporter/config.go#L17
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e2fb4ef305fb2f0de1e36f97e40b1ac0c33fdb4f/exporter/alertmanagerexporter/config.go#L21
... and many more

If we put this into a separate section, datadog exporter will indeed become an outlier